ci/gpu/build.sh

```diff
@@ -202,11 +202,26 @@ else
     conda list --show-channel-urls
 
     gpuci_logger "GoogleTests"
+
+    # Set up library for finding incorrect default stream usage.
+    cd "$WORKSPACE/cpp/tests/utilities/identify_stream_usage/"
+    mkdir build && cd build && cmake .. -GNinja && ninja && ninja test
+    STREAM_IDENTIFY_LIB="$WORKSPACE/cpp/tests/utilities/identify_stream_usage/build/libidentify_stream_usage.so"
+
     # Run libcudf and libcudf_kafka gtests from libcudf-tests package
     for gt in "$CONDA_PREFIX/bin/gtests/libcudf"*/* ; do
         test_name=$(basename ${gt})
+
         echo "Running GoogleTest $test_name"
-        ${gt} --gtest_output=xml:"$WORKSPACE/test-results/"
+        if [[ ${test_name} == "SPAN_TEST" ]]; then
+            # This one test is specifically designed to test using a thrust device
+            # vector, so we expect and allow it to include default stream usage.
```

Comment:
I agree. My thinking is that the approach in this PR is a bit of a stopgap until I can implement #11943 (which will take significantly longer since it depends on the API changes to enable streams). At that point we will probably want to implement a more surgical approach to skipping the check for specific tests or determining more precisely what streams to allow or prevent usage of.

Reply:
@davidwendt I chatted with @jrhemstad earlier today and he pointed out that this may become a moot point in the not too distant future since [spans will soon be in libcudacxx](https://github.com/NVIDIA/libcudacxx/pull/313) and we will eventually want to drop our spans in favor of those. That being the case, I've made a note in that test as you requested, but I don't think it's worth the effort to do anything else.